File: 01-week-1--starter-algorithms/03-day-4--selection-sort/ruby/selection_sort.rb
def selection_sort(arr)
  # type your code in here
  sorted_array = []
  
  until arr.length == 0
  min = arr.min 
  idx = arr.index(min)
  sorted_array.push(min)
  arr.delete_at(idx) 
  end
  
  return sorted_array
  
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort([3, -1, 5, 2])

  puts

  # Don't forget to add your own!

  # BENCHMARK HERE, you can print the average runtime
  long_input = []

  100.times { long_input << rand }
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 01-week-1--starter-algorithms/03-day-4--selection-sort/javascript/selection_sort.js
function selectionSort(arr) {
  // type your code here
  const sorted = []
  while (arr.length < 0){
    min = Math.min(...arr)
    idx = arr.indexOf(min)
    sorted.push(min)
    arr.spice
  }
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSort([3, -1, 5, 2]));

  console.log("");

  // BENCHMARK HERE, and print the average runtime
  const longInput = [];

  for (let i = 0; i < 100; ++i) {
    longInput.push(Math.random());
  }
}

module.exports = selectionSort;

// Please add your pseudocode to this file
/*****************************************
 * initalize with an empty array called sorted
 * 
 * Loop array length times 
 *  store minimum value in array in min 
 * remove minimum value from input array 
 * 
 * return sorted 
 *****************************************/
// And a written explanation of your solution
